src/WallsComputation.cpp

```diff
@@ -161,4 +170,45 @@ void WallsComputation::generateInsets(SliceLayer* layer)
     }
 }
 
+void WallsComputation::generateSpiralInsets(SliceLayerPart *part, coord_t line_width_0, coord_t wall_0_inset, bool recompute_outline_based_on_outer_wall)
+{
+    part->spiral_insets.push_back(part->outline.offset(-line_width_0 / 2 - wall_0_inset));
+
+    const size_t inset_part_count = part->spiral_insets[0].size();
+    constexpr size_t minimum_part_saving = 3; //Only try if the part has more pieces than the previous inset and saves at least this many parts.
+    constexpr coord_t try_smaller = 10; //How many micrometres to inset with the try with a smaller inset.
+    if (inset_part_count > minimum_part_saving + 1)
+    {
+        //Try a different line thickness and see if this fits better, based on these criteria:
+        // - There are fewer parts to the polygon (fits better in slim areas).
+        // - The polygon area is largely unaffected.
+        Polygons alternative_inset;
+        alternative_inset = part->outline.offset(-(line_width_0 - try_smaller) / 2 - wall_0_inset);
+
+        if (alternative_inset.size() < inset_part_count - minimum_part_saving) //Significantly fewer parts (saves more than 3 parts).
+        {
+            part->spiral_insets[0] = alternative_inset;
+        }
+    }
```

Comment:
This part here is meant to reduce the effect of the predecessor of the "wobbly lines" issue (currently in the sprint), where a slightly alternating polygon width could cause a wall to split up into many tiny little contours, because it alternates between slightly over 1 wall width and slightly under 1 wall width. This code tries a few line widths to see which leads to the smallest number of paths and chooses that then.

I think this is not necessary for spiralize mode. There's only 1 wall, so it would only apply for parts of around the thickness of 1 outer wall width. Spiralize doesn't deal well with thin walls anyway, especially now that wall overlap compensation is removed. So I think to make this code simpler, it's best to remove this.

Reply:
But spiralize reduces the entire shape to only its outer polygon. So technically, if the outer polygon becomes smaller by an outer wall line width, it may still still result in a multi-part polygon, which is not ideal for spiralization, considering that only the first part will always be spiralized. So it _may_ then still be a problem then, right?

Although I do understand that the possibility of such a case occuring is tremendously low.

Reply:
Yeah that's true, it would only apply for parts of around the thickness of 1 outer wall width. Those could get broken up in this case. Spiralize doesn't deal well with thin walls anyway due to its nature of gradually increasing the Z coordinate, especially now that wall overlap compensation is removed.

Also consider that this only tries 10 micrometres smaller line widths. This is intended to guard against rounding errors in the insets. The chance that it'll help with a very thin part is going to be very rare.